xorca/lib.py

```diff
@@ -58,7 +58,17 @@ def trim_and_squeeze(ds,
         ds = ds.isel(y=slice(*y_slice))
     if (x_slice is not None) and ("x" in ds.dims):
         ds = ds.isel(x=slice(*x_slice))
-    ds = ds.squeeze()
+    
+    to_squeeze = []
+    for dimensions in ds.dims:
+        if ds[dimensions].size == 1:
+            to_squeeze.append(dimensions)
+    
+    if "t" in to_squeeze:    
+        t_index = to_squeeze.index("t")
+        del to_squeeze[t_index]
+        
+    ds = ds.squeeze(dim = to_squeeze)
```

Comment:
We should to this with a list comprehension:

```python
to_squeeze = [dim for dim in ds.dims if ((ds[dim].size == 1) and (dim is not "t")]
```

And we should make sure that this is tested.  Let's include a singleton dimension "t" here https://github.com/willirath/xorca/blob/0447e26428ed5f75ae5711b697d0562d7c6f5646/xorca/tests/test_lib.py#L24 and `assert` that it's still present in the resulting data set.